actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -1586,6 +2056,229 @@ def fields_for(record_name, record_object = nil, fields_options = {}, &block)
+      #   end
+      #
+      # This model can now be used with a nested +fields+. The block given to
+      # the nested +fields+ call will be repeated for each instance in the
+      # collection:
+      #
+      #   <%= form_with model: @person do |person_form| %>
+      #     ...
+      #     <%= person_form.fields :projects do |project_fields| %>
+      #       <% if project_fields.object.active? %>
+      #         Name: <%= project_fields.text_field :name %>
+      #       <% end %>
+      #     <% end %>
+      #     ...
+      #   <% end %>
+      #
+      # It's also possible to specify the instance to be used:
+      #
+      #   <%= form_with model: @person do |person_form| %>
+      #     ...
+      #     <% @person.projects.each do |project| %>
+      #       <% if project.active? %>
+      #         <%= person_form.fields :projects, model: project do |project_fields| %>
+      #           Name: <%= project_fields.text_field :name %>
+      #         <% end %>
+      #       <% end %>
+      #     <% end %>
+      #     ...
+      #   <% end %>
+      #
+      # Or a collection to be used:
+      #
+      #   <%= form_with model: @person do |person_form| %>
+      #     ...
+      #     <%= person_form.fields :projects, model: @active_projects do |project_fields| %>
+      #       Name: <%= project_fields.text_field :name %>
+      #     <% end %>
+      #     ...
+      #   <% end %>
+      #
+      # To destroy any of the associated models through the
+      # form, you have to enable it using <tt>:allow_destroy</tt>:
+      #
+      #   class Person < ActiveRecord::Base
+      #     has_many :projects
+      #     accepts_nested_attributes_for :projects, allow_destroy: true
+      #   end
+      #
+      # This will allow you to specify which models to destroy in the
+      # attributes hash by adding a form element for the <tt>_destroy</tt>
+      # parameter with a value that evaluates to +true+
+      # (eg. 1, '1', true, or 'true'):
+      #
+      #   <%= form_with model: @person do |person_form| %>
+      #     ...
+      #     <%= person_form.fields :projects do |project_fields| %>
+      #       Delete: <%= project_fields.check_box :_destroy %>
+      #     <% end %>
+      #     ...
+      #   <% end %>
+      #
+      # When collections is used you might want to know the index of each
+      # object into the array. Use the FormBuilder's <tt>index</tt> method:
+      #
+      #   <%= form_with model: @person do |person_form| %>
+      #     ...
+      #     <%= person_form.fields :projects do |project_fields| %>
+      #       Project #<%= project_fields.index %>
+      #       ...
+      #     <% end %>
+      #     ...
+      #   <% end %>
+      #
+      # Note +fields+ automatically generates a hidden field to store the record
+      # ID. For circumstances where this is not needed pass
+      # <tt>include_id: false</tt> to skip it.
+      def fields(scope = nil, model: nil, **options, &block)
```

Comment:
Yup, copied to mirror how `fields_for` documents itself (i.e. both as the helper and in the builder). I'll update this to just refer to the `fields` helper docs above.